compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -628,7 +628,14 @@ class SpaceEngine(using Context) extends SpaceLogic {
       case tp if tp.classSymbol.isAllOf(JavaEnumTrait) =>
         tp.classSymbol.children.map(sym => Typ(sym.termRef, true))
       case tp =>
-        val children = tp.classSymbol.children
+        def getChildren(sym: Symbol): List[Symbol] =
+          sym.children.flatMap { child =>
+            if child eq sym then Nil // i3145: sealed trait Baz, val x = new Baz {}, Baz.children returns Baz...
+            else if tp.classSymbol == defn.TupleClass || tp.classSymbol == defn.NonEmptyTupleClass then List(child) // Tuple is weird/different...
```

Comment:
```suggestion
            else if tp.classSymbol == defn.TupleClass || tp.classSymbol == defn.NonEmptyTupleClass then List(child) // Tuple types are substituted for different classes at runtime
```

Reply:
Actually, it makes me think: is there a reason TupleXXL doesn't extend Tuple and the TupleN classes made synthetically to extend Tuple?  Then we wouldn't need this.